llm/llm.go

```diff
@@ -27,6 +30,34 @@ func New(model string, opts api.Options) (LLM, error) {
 		return nil, err
 	}
 
+	switch ggml.FileType {
+	case FileTypeQ5_0, FileTypeQ5_1, FileTypeQ8_0:
+		// Q5_0, Q5_1, and Q8_0 do not support Metal API and will
+		// cause the runner to segmentation fault so disable GPU
+		log.Printf("WARNING: GPU disabled for Q5_0, Q5_1, and Q8_0")
+		opts.NumGPU = 0
+	}
+
+	totalResidentMemory := memory.TotalMemory()
+	switch ggml.ModelType {
+	case ModelType3B, ModelType7B:
+		if totalResidentMemory < 8*1024*1024 {
+			return nil, fmt.Errorf("model requires at least 8GB of memory")
```

Comment:
These could be warnings instead but they must be visible client-side

Reply:
Was going to mention that: better if we fallback to CPU?

Great point on notifying the client too

Reply:
I'm not sure that'll be a good experience since trying to load a too-large model will either crash ollama or lead to very bad (unusable) performance.

Reply:
This is also a very rough estimate. There are still other ways for the system to OOM, such as using a too large context

Reply:
can we exit but allow people to override with some kind of force flag?